src/sections/Meshery/Meshery-integrations/index.js

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { useState } from "react";
+import IntegrationsGrid from "./IntegrationsGrid";
+import IntegrationList from "../../../assets/data/integrations/IntegrationList";
+import Integrations from "./Integration.style";
+
+function index() {
+  const [categoryList, setcategoryList] = useState([
+    { name: "All", isSelected: true },
+    { name: "Platforms", isSelected: false },
+    { name: "Service Mesh", isSelected: false },
+    { name: "Operating System", isSelected: false },
+    { name: "Collaboration", isSelected: false },
+  ]);
+
+  const [integrationCollection, setintegrationCollection] =
+    useState(IntegrationList);
+
+  const filterCategory = (e) => {
+    const category = e.target.innerHTML;
+
+    if (category !== "All") {
+      categoryList[0].isSelected = false;
+      categoryList.map((item) =>
+        item.name === category ? (item.isSelected = !item.isSelected) : null
+      );
+      setcategoryList(categoryList);
+    } else {
+      categoryList[0].isSelected = true;
+
+      categoryList.map((item) =>
+        item.name !== "All" ? (item.isSelected = false) : null
+      );
+      setcategoryList(categoryList);
+    }
+
+    let tempCat = [];
+
+    categoryList.map((item) => {
+      if (item.isSelected) tempCat.push(item.name);
```

Comment:
Move the statement to next line

Reply:
Also, when you are just looping through the list why use `map`?
Simply use `forEach`

Reply:
And also, all other places where it is not returning anything and just looping through the data. Always use `forEach`

Reply:
Okay.